### ISSUE TITLE
FilePulse Connector adding additional information to the message

### ISSUE THREAD

**Issue by dmarupov** [streamthoughts/kafka-connect-file-pulse#67]
I am getting an additional information on the message that is being sent from the source. For example:
The original message is: `{"firstname": "John", "lastname": "Doe"}`
After FilePulse processes the message it becomes like this: `Struct{message={"firstname": "John", "lastname": "Doe"}}`

So FilePulse is adding `'Struct{message=}'` is this an expected behavior? If so is there a way to remove it or truncate it before the message gets to the topic? Is it needed?

Thank you for you help!

**Comment by fhussonnois** [comment]
Hi @dmarupov, did you configure filters (i.e JsonFilter)? 

If you don't use filter then this is the expected behavior. FilePulse creates a single field `message` with the content of the entire file or a single line of the file.

**Comment by dmarupov** [comment]
Hi @fhussonnois,

I tried to test with JsonFilter real quick and I got the same result. Below is the configuration I used for FilePulse:
connector.class=io.streamthoughts.kafka.connect.filepulse.source.FilePulseSourceConnector
```
filters=MyJsonFilter
filters.MyJsonFilter.source=message
filters.MyJsonFilter.type=io.streamthoughts.kafka.connect.filepulse.filter.JSONFilter
fs.cleanup.policy.class=io.streamthoughts.kafka.connect.filepulse.clean.LogCleanupPolicy
fs.scan.directory.path=/mnt/filepulse
internal.kafka.reporter.bootstrap.servers=my.ip.address:9091,my.ip.address:9092,my.ip.address:9093
key.converter=org.apache.kafka.connect.storage.StringConverter
name=filepulse-test
tasks.max=3
topic=mytopic
value.converter=org.apache.kafka.connect.storage.StringConverter
```


We were also using the below filter for our dev environment:
```
filters=Drop
filters.Drop.if={{ matches($value.message, .*"File": "(FIRSTNAME|MIDDLE|LASTNAME|ADDRESS)".*) }}
filters.Drop.invert=true
filters.Drop.type=io.streamthoughts.kafka.connect.filepulse.filter.DropFilter
```

**Comment by fhussonnois** [comment]
Yes, the `JSONFilter` doesn't remove the field `message`. You have to remove it. 

So, using Connect FilePulse 1.5.0 you should configure your connector like this : 

```
filters=Drop, ParseJson, DropMessageField
filters.Drop.type=io.streamthoughts.kafka.connect.filepulse.filter.DropFilter
filters.Drop.if={{ matches($value.message, .*"File": "(FIRSTNAME|MIDDLE|LASTNAME|ADDRESS)".*) }}
filters.Drop.invert=true
filters.ParseJson.type=io.streamthoughts.kafka.connect.filepulse.filter.JSONFilter
filters.ParseJson.source=message
filters.ParseJson.merge=true
filters.DropMessageField.type=io.streamthoughts.kafka.connect.filepulse.filter.ExcludeFilter
filters.DropMessageField.fields=message
```

**Comment by dmarupov** [comment]
Thank you I will try that.

**Comment by dmarupov** [comment]
Hi @fhussonnois,
I installed the Connect FilePulse 1.5.0 and used the following configuration as you suggested:
```
connector.class=io.streamthoughts.kafka.connect.filepulse.source.FilePulseSourceConnector
filters=Drop, ParseJson, ExcludeMessageField
filters.Drop.if={{ matches($value.message, .*"File": "(FIRSTNAME|MIDDLE|LASTNAME|ADDRESS)".*) }}
filters.Drop.invert=true
filters.Drop.type=io.streamthoughts.kafka.connect.filepulse.filter.DropFilter
filters.ExcludeMessageField.fields=message
filters.ExcludeMessageField.type=io.streamthoughts.kafka.connect.filepulse.filter.ExcludeFilter
filters.ParseJson.merge=true
filters.ParseJson.source=message
filters.ParseJson.type=io.streamthoughts.kafka.connect.filepulse.filter.JSONFilter
fs.cleanup.policy.class=io.streamthoughts.kafka.connect.filepulse.clean.LogCleanupPolicy
fs.scan.directory.path=/mnt/filepulse
header.converter=org.apache.kafka.connect.storage.StringConverter
internal.kafka.reporter.bootstrap.servers=my.ip.address:9092,my.ip.address:9093,my.ip.address:9094
key.converter=org.apache.kafka.connect.storage.StringConverter
name=FilePulseMain
tasks.max=3
topic=mytopic
value.converter=org.apache.kafka.connect.storage.StringConverter
```
And I am getting the following error:
```
org.apache.kafka.connect.errors.ConnectException: io.streamthoughts.kafka.connect.filepulse.expression.parser.ScELParseCancellationException: Cannot parse ScEL expression '{{ matches($value.message, .*"File": "(FIRSTNAME|MIDDLE|LASTNAME|ADDRESS)".*) }}' : line 1:27 mismatched input '.' expecting Literal
	at io.streamthoughts.kafka.connect.filepulse.config.TaskConfig.filters(TaskConfig.java:155)
	at io.streamthoughts.kafka.connect.filepulse.source.FilePulseSourceTask.newFileRecordsPollingConsumer(FilePulseSourceTask.java:89)
	at io.streamthoughts.kafka.connect.filepulse.source.FilePulseSourceTask.start(FilePulseSourceTask.java:81)
	at org.apache.kafka.connect.runtime.WorkerSourceTask.execute(WorkerSourceTask.java:213)
	at org.apache.kafka.connect.runtime.WorkerTask.doRun(WorkerTask.java:184)
	at org.apache.kafka.connect.runtime.WorkerTask.run(WorkerTask.java:234)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: io.streamthoughts.kafka.connect.filepulse.expression.parser.ScELParseCancellationException: Cannot parse ScEL expression '{{ matches($value.message, .*"File": "(FIRSTNAME|MIDDLE|LASTNAME|ADDRESS)".*) }}' : line 1:27 mismatched input '.' expecting Literal
	at io.streamthoughts.kafka.connect.filepulse.expression.parser.antlr4.Antlr4ExpressionParser$ThrowingErrorListener.syntaxError(Antlr4ExpressionParser.java:97)
	at org.antlr.v4.runtime.ProxyErrorListener.syntaxError(ProxyErrorListener.java:41)
	at org.antlr.v4.runtime.Parser.notifyErrorListeners(Parser.java:544)
	at org.antlr.v4.runtime.DefaultErrorStrategy.reportInputMismatch(DefaultErrorStrategy.java:327)
	at org.antlr.v4.runtime.DefaultErrorStrategy.reportError(DefaultErrorStrategy.java:139)
	at io.streamthoughts.kafka.connect.filepulse.expression.parser.antlr4.ScELParser.value(ScELParser.java:809)
	at io.streamthoughts.kafka.connect.filepulse.expression.parser.antlr4.ScELParser.functionParameters(ScELParser.java:720)
	at io.streamthoughts.kafka.connect.filepulse.expression.parser.antlr4.ScELParser.functionDeclaration(ScELParser.java:656)
	at io.streamthoughts.kafka.connect.filepulse.expression.parser.antlr4.ScELParser.expression(ScELParser.java:434)
	at io.streamthoughts.kafka.connect.filepulse.expression.parser.antlr4.ScELParser.substitutionStrExpression(ScELParser.java:363)
	at io.streamthoughts.kafka.connect.filepulse.expression.parser.antlr4.ScELParser.substitutionExpression(ScELParser.java:242)
	at io.streamthoughts.kafka.connect.filepulse.expression.parser.antlr4.ScELParser.scel(ScELParser.java:163)
	at io.streamthoughts.kafka.connect.filepulse.expression.parser.antlr4.Antlr4ExpressionParser.parseExpression(Antlr4ExpressionParser.java:72)
	at io.streamthoughts.kafka.connect.filepulse.expression.parser.antlr4.Antlr4ExpressionParser.parseExpression(Antlr4ExpressionParser.java:57)
	at io.streamthoughts.kafka.connect.filepulse.expression.parser.ExpressionParsers.parseExpression(ExpressionParsers.java:37)
	at io.streamthoughts.kafka.connect.filepulse.filter.condition.ExpressionFilterCondition.<init>(ExpressionFilterCondition.java:44)
	at io.streamthoughts.kafka.connect.filepulse.config.CommonFilterConfig.condition(CommonFilterConfig.java:85)
	at io.streamthoughts.kafka.connect.filepulse.filter.AbstractRecordFilter.configure(AbstractRecordFilter.java:50)
	at io.streamthoughts.kafka.connect.filepulse.config.TaskConfig.filters(TaskConfig.java:152)
	... 10 more
```

I also tried to remove the Drop Filter just to see and it did process but it still includes the word `Struct`. An example is shown below:
`Struct{"firstname": "John", "lastname": "Doe"}`

**Comment by dmarupov** [comment]
I also notices that everytime there is a nested JSON it adds the word `Struct` in front of it. Like so:
```
**Struct**{
	fistname=some info,
	lastname=some info,
	address=**Struct**{
			street=some info,
			city=some info,
			state=some info
		}
	}
```

**Comment by dmarupov** [comment]
Hi @fhussonnois,
I also tried to remove all the filters and process some data. The word `Struct` still being added. I would really like to be able to process the messages as they are without adding any additional data to it. Is it possible with FilePulse?

Thank you very much!

**Comment by fhussonnois** [comment]
Hi @dmarupov, I'm sorry I've missed to fix your expression; here is the good one (note the quotes) : 

```
{{ matches($value.message, '.*"File": "(FIRSTNAME|MIDDLE|LASTNAME|ADDRESS)".*)' }}
```

Which serializer are you using ? Kafka Connect uses its own internal class `Struct` which represent a data record. This the `toString` method output that you are giving me not the serialized record send to Kafka. See https://github.com/apache/kafka/blob/12f310d50e7f5b1c18c4f61a119a6cd830da3bc0/connect/api/src/main/java/org/apache/kafka/connect/data/Struct.java#L268

**Comment by dmarupov** [comment]
Hi @fhussonnois,
Thank you for your suggestions, I really appreciate them. 
Just an FYI, I ended up using `org.apache.kafka.connect.json.JsonConverter` for `value.converter` and this got rid of the word `Struct` which solves my issue. 
Thank you very much for all of your help!

**Comment by aonamrata** [comment]
Even after 5 years, this still is relevant. I stumbled across this same issue and ☝️ solutoin worked. The only other catch in my case was that I had to specify both explicitly 
```
    "value.converter": "org.apache.kafka.connect.json.JsonConverter",
    "value.converter.schemas.enable": "false",
```
for it to work and not write the `schema`, even though I had `ENV CONNECT_VALUE_CONVERTER='org.apache.kafka.connect.json.JsonConverter'` and was getting json messages without it.
